modules/programs/keepassxc.nix

```diff
@@ -43,16 +46,27 @@ in
         for the full list of options.
       '';
     };
+
+    autostart = lib.mkOption {
+      type = lib.types.bool;
+      default = false;
+      example = true;
+      description = ''
+        Whether to start Keepassxc automatically on login through the XDG autostart mechanism.
+        Make sure to also enable {option}`xdg.autostart.enable`.
```

Comment:
Actually, reminds me that we should have an assertion that enabling this requires that setting to be enabled.

Reply:
But what's the purpose of the `xdg.autostart.enable` flag then? I thought it is used to disable autostarting altogether.

Reply:
We don't want unexpected side effects from a module, though. So, a warning/assertion is preferred so a user can explicitly opt in to a behavior from elsewhere. Just enabling this module's `autostart` doesn't mean I expect any other `xdg.autostart` mechanisms to be enabled that might be configured in a configuration. Similar to how we have configured most other cross module integrations.

Reply:
>  Is there any possibility to add a test for it?

There are a handful of tests that exist to test assertions, you could just create another test case matching their strategy of expecting an assertion thrown with a config.

Reply:
Ok, should be done now.